### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by MggMuggins** [Sinthorion/lotr-minecraft-mod-framework#2]
Alright, so you have this section on your Readme:

Progress/Goals
--------------
Current version: early alpha stage

- [ ] World generation
  - [ ] Map file
  - [ ] Biome list
  - [ ] WorldType
  - [ ] Custom biomes
  - [ ] Waypoints
  - [ ] Roads
  - [ ] Worldgen features
- [ ] Items
  - [ ] Crafting tables
  - [ ] Banners
  - [ ] Horse Armour
  - [ ] Brewables
  - [ ] Chest loot
  - [ ] ...
- [ ] Factions
- [ ] NPCs
- [ ] Achievements
- [ ] ...

How are we planning to implement all this stuff? What is the overall goal for each of these sections? It might be good to look at some users of the mod (the Server team you're working with (btw, who is in that group?), or other people looking to customize the mod in other ways) and see what they would like to see in the end product, and we can kind of build our software around that.

**Comment by Sinthorion** [comment]
As far as I understand it, the focus lies on world generation, especially things like roads and waypoints, because that are real issues for a submod that modifies the map. Everything else can be emulated to a great extend by retexturing and relabeling the existing mods, at least for Middle Earth based worlds.

Another front is the interface through which the custom world configurations are supplied. At first I would keep it purely as an API, requiring another submod to actually change stuff, but mid-term I would like a non-Java configuration option as well, optimally through JSON files from a resource pack, if that's possible.

**Comment by MggMuggins** [comment]
Right, so that requires a pretty complete knowledge of how Mevans does WorldGen to even begin writing anything that would be useful. Which right now, that boils down to exactly one person...

API is a good starting goal. JSON and configuration are hard to do, because you almost have to put everything you want into the config options, and it's simply not as flexible.
